drivers/pwm/pwm_nrfx.c

```diff
@@ -121,6 +122,98 @@ static int pwm_nrfx_init(struct device *dev)
 	return 0;
 }
 
+#ifdef CONFIG_DEVICE_POWER_MANAGEMENT
+static void pwm_nrfx_set_power_state(u32_t new_state, struct device *dev)
+{
+	const struct pwm_nrfx_config *pconfig = dev->config->config_info;
+
+	if (new_state == DEVICE_PM_ACTIVE_STATE) {
+		nrfx_err_t result = nrfx_pwm_init(&pconfig->pwm,
+						&pconfig->config, NULL);
+
+		if (result != NRFX_SUCCESS) {
+			SYS_LOG_ERR("Failed to initialize device: %s",
+				    dev->config->name);
+		}
+
+		nrfx_pwm_simple_playback(&pconfig->pwm,
+				 &pconfig->seq,
+				 1,
+				 NRFX_PWM_FLAG_LOOP);
+	} else {
+		assert(new_state == DEVICE_PM_LOW_POWER_STATE ||
+		       new_state == DEVICE_PM_SUSPEND_STATE ||
+		       new_state == DEVICE_PM_OFF_STATE);
+
+		nrfx_pwm_uninit(&pconfig->pwm);
+	}
+}
+
+static int pwm_nrfx_pm_control(struct device *dev,
+				u32_t ctrl_command,
+				void *context)
+{
+	static u32_t current_state[] = {
+#ifdef CONFIG_PWM_3
+		DEVICE_PM_ACTIVE_STATE, DEVICE_PM_ACTIVE_STATE,
+		DEVICE_PM_ACTIVE_STATE, DEVICE_PM_ACTIVE_STATE
+#elif defined CONFIG_PWM_2
+		DEVICE_PM_ACTIVE_STATE, DEVICE_PM_ACTIVE_STATE,
+		DEVICE_PM_ACTIVE_STATE
+#elif defined CONFIG_PWM_1
+		DEVICE_PM_ACTIVE_STATE, DEVICE_PM_ACTIVE_STATE
+#elif defined CONFIG_PWM_0
+		DEVICE_PM_ACTIVE_STATE
+#endif
+	};
+
+	static const char *dev_name[] = {
+#ifdef CONFIG_PWM_0
+		CONFIG_PWM_0_NAME,
+#else
+		"",
+#endif
+#ifdef CONFIG_PWM_1
+		CONFIG_PWM_1_NAME,
+#else
+		"",
+#endif
+#ifdef CONFIG_PWM_2
+		CONFIG_PWM_2_NAME,
+#else
+		"",
+#endif
+#ifdef CONFIG_PWM_3
+		CONFIG_PWM_3_NAME,
+#endif
+	};
+
+	int dev_index = -1;
+
+	for (size_t i = 0; i < ARRAY_SIZE(dev_name); i++) {
+		if (!strcmp(dev->config->name, dev_name[i])) {
```

Comment:
since these are const pointers on flash there will be only one copy of it so you can simply compare pointers.

Reply:
State is now in device data structure